### PR TITLE
Lint: Allow params.reads or params.input or params.design

### PR DESCRIPTION
**Pull request by ewels**
- Also refactored a bit of the lint code for files and config options
- Also made some of the output a little more colourful and pretty :sparkles:

Fixes https://github.com/nf-core/tools/issues/497

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

### CODE REVIEW

**Comment by ewels** [comment]
Before:

image

After:

image

File: docs/lint_errors.md [hunk 1]
```diff
@@ -100,8 +100,12 @@ The following variables throw warnings if missing:
...
   * Must specify a tag that matches the pipeline version number if set.
   * If the pipeline version number contains the string `dev`, the DockerHub tag must be `:dev`
-* `params.reads`
-  * Input parameter to specify input data (typically FastQ files / pairs)
+* `params.reads` or `params.input` or `params.design`
+  * Input parameter to specify input data - one or more of these can be used to avoid a warning
+  * Typical usage:
+    * `params.reads`: FastQ files (or pairs)
+    * `params.input`: Input data that is not NGS sequencing data
+    * `params.design`: A CSV design file specifying input files and metadata for the run
```
Comment (MaxUlysse) [inline docs/lint_errors.md]:
```suggestion
    * `params.design`: A CSV/TSV design file specifying input files and metadata for the run
```

Comment (MaxUlysse) [reply docs/lint_errors.md]: we actually use TSV files in Sarek

File: nf_core/lint.py [hunk 2]
```diff
@@ -226,25 +226,27 @@ def check_files_exist(self):
         """
 
         # NB: Should all be files, not directories
-        # Supplying a list means if any are present it's a pass
+        # Supplying a list means if any are present it's a pass - lists
```
Comment (MaxUlysse) [inline nf_core/lint.py]: Not sure that the comments is more clear with the extra ` - lists`

Comment (ewels) [reply nf_core/lint.py]: The comment below says `- strings`. It's just a note to future me that this should be a list of lists, and the other a list of strings. It wasn't very important, but yes I agree it's not super clear.

Comment (ewels) [reply nf_core/lint.py]:
```suggestion
        # List of lists. Passes if any of the files in the sublist are found.
```

File: nf_core/lint.py [hunk 3]
```diff
@@ -226,25 +226,27 @@ def check_files_exist(self):
...
         files_warn = [
-            'main.nf',
-            'environment.yml',
-            os.path.join('conf','base.config')
+            ['main.nf'],
+            ['environment.yml'],
+            [os.path.join('conf','base.config')]
         ]
+
+        # These trigger fails / warns if any exist - strings
```
Comment (ewels) [inline nf_core/lint.py]:
```suggestion
        # List of strings. Dails / warns if any of the strings exist.
```

File: nf_core/lint.py (hunk 2 above)
Comment (MaxUlysse) [reply nf_core/lint.py]: I'm usually looking at Python without understanding much, and this was even less that usual, so I was concerned, but if you understand what you meant, it's good enough I think

**Pull request merged**